src/entity/terrain_mesh.cpp

```diff
@@ -2,98 +2,50 @@
 #include "terrain_mesh.hpp"
 
 #include "../logging.hpp"
+//#include "../terrain/material.hpp"
 
 #include <GL/glew.h>
 #include <GLFW/glfw3.h>
 #include <glm/glm.hpp>
 
-namespace terrain {
-
-void
-TerrainMesh::init(entity::Mesh mesh) {
-    init(
-        mesh.get_indices(), mesh.get_indexed_vertices(), mesh.get_indexed_color_ids(),
-        mesh.get_indexed_normals(), mesh.get_color_map()
-    );
+terrain::TerrainMesh::TerrainMesh(entity::Mesh mesh) {
+    init(mesh);
```

Comment:
Just move `init` into your ctor at this point

Reply:
Delete init

Put body in ctor

Reply:
But I use init.

Reply:
"A designer knows he has achieved perfection not when there is nothing left to add, but when there is nothing left to take away."

You can take away the init function body an inline it into the constructor. Do it, it'll be faster and easier to maintain. You don't call `init` anywhere else.

Reply:
ohh boy you don't know about `std::move` do you?

Ping me on discord and I'll explain.